firewood/src/merkle/node.rs

```diff
@@ -490,6 +495,93 @@ pub enum EncodedNodeType {
     },
 }
 
+// TODO: probably can merge with `EncodedNodeType`.
+#[derive(Debug, Deserialize)]
+struct EncodedBranchNode {
+    chd: Vec<(u64, Vec<u8>)>,
+    data: Option<Vec<u8>>,
+    path: Vec<u8>,
+}
+
+// Note that the serializer passed in should always be the same type as T in EncodedNode<T>.
+impl Serialize for EncodedNode<PlainCodec> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let n = match &self.node {
+            EncodedNodeType::Leaf(n) => {
+                let data = Some(n.data.to_vec()).filter(|d| !d.is_empty());
```

Comment:
You should be able to store empty data

Reply:
Yeah I guess empty value is different from zero value in this case. Tried to match with MerkleDB but I think your suggestion is actually right with a second thought.